dangerzone/document.py

```diff
@@ -12,9 +12,9 @@ class Document(object):
     The Common class is a singleton of shared functionality throughout an open dangerzone window
     """
 
-    def __init__(self) -> None:
+    def __init__(self, input_file_path: str) -> None:
```

Comment:
The "simplify 'new window' logic" commit tries to do three things at once:

1. Change the way we instantiate the document class.
2. Moves the filename checks out of the new window function.
3. Simplifies the window logic a bit.

Specifically for (2), we move this logic again under the `DocumentHolder()` class in a subsequent commit. So, all in all, it's better to break these three things in different commits, and merge them with any amendments we do in subsequent commits. This way, each commit has a clear purpose, and it doesn't introduce logic that will be altered.

Reply:
Great, thanks for the explanation.